CAToolsetPerfDataCollector/PerfDataCollector.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <iostream>
+#include <CppBuildInsights.hpp>
+#include <map>
+#include <unordered_map>
+#include <compare>
+
+#include <comutil.h>
+
+// Please refer to main.cpp file for terms / acronyms used in this file.
+
+using namespace Microsoft::Cpp::BuildInsights;
+using namespace Activities;
+using namespace SimpleEvents;
+using namespace std::chrono;
+
+struct PathId
+{
+    int id;
+    const std::wstring* path;
+
+    constexpr auto operator<=>(const PathId& b) const { return *path <=> *b.path; };
```

Comment:
Would ordering based on `id` make sense here?

Reply:
It really doesn't make difference if the records are to be processed through another tool. Just wanted to group sources from same directories to be grouped together, instead of the order in which they appear in the log file. For now, it seems like this more costly comparison is acceptable for reasonable size logs (about 10GB or so logs). If it gets in the way, we can change. Leaving it as is for now.